addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -165,6 +166,78 @@ def map_account_tax_template_type(cr):
         table='account_tax_template', write='sql')
 
 
+def precreate_fields(cr):
+    """Create computed fields that take long time to compute, but will be
+    filled with valid values by migration."""
+
+    def create_field(cr, table_name, field_name, pg_type, comment):
+        cr.execute(
+            'ALTER TABLE "%s" ADD COLUMN "%s" %s' %
+            (table_name, field_name, pg_type)
+        )
+        cr.execute(
+            'COMMENT ON COLUMN %s."%s" IS %%s' % (table_name, field_name),
+            (comment,)
+        )
+
+    create_field(
+        cr,
+        "account_move", "currency_id", "int4",
+        "Currency"
+    )
+    create_field(
+        cr,
+        "account_move", "amount", "numeric",
+        "Amount"
+    )
+    create_field(
+        cr,
+        "account_move", "matched_percentage", "numeric",
```

Comment:
note that everything you create here will keep the value `NULL` if you don't fill the value manually, I think this is the case with this column and some below too.
Also note that when you precreate a column, you're responsible for eventual indexes on the columns you create, if you don't create them too, the database might be slower than it should. But for the columns here, I don't think upstream sets indexes.